lib/loader.js

```diff
@@ -90,12 +86,44 @@ function loadDatabase(app, config) {
     initializer(factory) {
       if (typeof factory === 'function') {
         // module.exports = class User extends Bone {};
-        if (factory.prototype instanceof Model) return factory;
+        if (Realm.isBone(factory)) {
+          if (config.sequelize && !factory.sequelize) {
+            const name = factory.name;
+            const sequelized = Realm.sequelize(factory);
+
+            Object.defineProperties(sequelized, {
+              name: { value: name },
+              // should work with associations
+              models: {
+                get() {
+                  return Model.models;
+                }
+              },
+              app: {
+                get() {
+                  return app;
+                },
+                configurable: true,
+                enumerable: false,
+              }
+            });
+            Object.defineProperty(sequelized.prototype, 'app', {
+              get() {
+                return app;
+              },
+              configurable: true,
+              enumerable: false,
+            });
+
+            return sequelized;
```

Comment:
indent level gets too deep, might consider refactor it by preceding some early returns?